src/order/symm_diff.lean

```diff
@@ -164,21 +185,40 @@ lemma symm_diff_right_comm : a ∆ b ∆ c = a ∆ c ∆ b := by simp_rw [symm_d
 lemma symm_diff_symm_diff_symm_diff_comm : (a ∆ b) ∆ (c ∆ d) = (a ∆ c) ∆ (b ∆ d) :=
 by simp_rw [symm_diff_assoc, symm_diff_left_comm]
 
+lemma symm_diff_triangle : a ∆ c ≤ a ∆ b ⊔ b ∆ c :=
+begin
+  refine (sup_le_sup (sdiff_triangle a b c) $ sdiff_triangle _ b _).trans_eq _,
+  rw [@sup_comm _ _ (c \ b), sup_sup_sup_comm],
+  refl,
+end
+
 @[simp] lemma symm_diff_symm_diff_self : a ∆ (a ∆ b) = b := by simp [←symm_diff_assoc]
 
 @[simp] lemma symm_diff_symm_diff_self' : a ∆ b ∆ a = b :=
 by rw [symm_diff_comm, ←symm_diff_assoc, symm_diff_self, bot_symm_diff]
 
-@[simp] lemma symm_diff_right_inj : a ∆ b = a ∆ c ↔ b = c :=
-begin
-  split; intro h,
-  { have H1 := congr_arg ((∆) a) h,
-    rwa [symm_diff_symm_diff_self, symm_diff_symm_diff_self] at H1, },
-  { rw h, },
-end
+namespace equiv
+
+/-- Symmetric difference by `a` as an equivalence. We make `equiv.symm_diff a b` and
+`(equiv.symm_diff a).symm b` defeq to `a ∆ b` and `b ∆ a` for convenience. -/
+protected def symm_diff (a : α) : α ≃ α :=
+{ to_fun := (∆) a,
+  inv_fun := λ b, b ∆ a,
+  left_inv := symm_diff_symm_diff_self' _,
+  right_inv := λ b, by rw [←symm_diff_assoc, symm_diff_symm_diff_self'] }
+
+@[simp] lemma coe_symm_diff (a : α) : ⇑(equiv.symm_diff a) = (∆) a := rfl
+@[simp] lemma symm_diff_apply (a b : α) : equiv.symm_diff a b = a ∆ b := rfl
```

Comment:
Isn't this lemma already provable by simp? I'm slightly surprised the linter doesn't flag it

Reply:
How would it? There is no`@[simps]` on `equiv.symm_diff`.

Reply:
From the lemma immediately above

Reply:
Ah sure, this is the `coe_whatever`/`whatever_apply` pair. We have it for all homs because applied and non-applied lemmas behave differently in simp.

Reply:
Actually, why isn't there a `@[simps]` on `equiv.symm_diff`?